### PR TITLE
Modifying code examples to include training data

### PR DESCRIPTION
**Pull request by barneyhill**
Rewritten current code examples that rely on training data so that they are self-contained. This issue is mentioned at [#1049](https://github.com/optuna/optuna/issues/1049)

### CODE REVIEW

**Comment by barneyhill** [comment]
I'm quite confused about the failed example. Could someone help me find whereabouts in trial.py the test is failing? Thankyou.

**Comment by toshihikoyanase** [comment]
Hi @barneyhill, thank you for your PR!

> Could someone help me find whereabouts in trial.py the test is failing? 

In the error message, the `import optuna` line was not recognized as a part of example code:

```
File "../../venv/lib/python3.8/site-packages/optuna/trial.py", line ?, in import optuna
Failed example:
    import numpy as np

    from sklearn.neural_network import MLPClassifier
    from sklearn.datasets import load_iris
    from sklearn.model_selection import train_test_split

    X, y = load_iris(return_X_y=True)
    X_train, X_test, y_train, y_test = train_test_split(X, y)
    classes = np.unique(y)

    def objective(trial):
        momentum = trial.suggest_uniform('momentum', 0.0, 1.0)
        clf = MLPClassifier(hidden_layer_sizes=(100, 50), momentum=momentum,
                            solver='sgd', random_state=0)
        clf.fit(X_train, y_train)

        return clf.score(X_test, y_test)

    study = optuna.create_study(direction='maximize')
    study.optimize(objective, n_trials=3)
Exception raised:
    Traceback (most recent call last):
      File "/home/docs/.pyenv/versions/3.8.0/lib/python3.8/doctest.py", line 1328, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest import optuna[0]>", line 19, in <module>
        study = optuna.create_study(direction='maximize')
    NameError: name 'optuna' is not defined
```

I guess some problems happened in the code extractor. So, I re-formatted the import lines and all tests succeeded.

The actual changes are as follows:
- removed blank lines between import-lines 
- reordered import lines in alphabetical order

Example: 
```python
# before
                import optuna

                import numpy as np

                from sklearn.datasets import load_iris
                from sklearn.neural_network import MLPClassifier
                from sklearn.model_selection import train_test_split

# after
                import optuna
                import numpy as np
                from sklearn.datasets import load_iris
                from sklearn.model_selection import train_test_split
                from sklearn.neural_network import MLPClassifier
```

**Comment by barneyhill** [comment]
@toshihikoyanase thanks a lot for the help!

File: optuna/exceptions.py [hunk 1]
```diff
@@ -13,22 +13,19 @@ class TrialPruned(OptunaError):
...
 
-        .. testsetup::
+        .. testcode::
 
+            import optuna
             import numpy as np
+            from sklearn.linear_model import SGDClassifier
+
+            from sklearn.datasets import load_iris
             from sklearn.model_selection import train_test_split
```
Comment (toshihikoyanase) [inline optuna/exceptions.py]: Please remove the unnecessary blank line and sort lines in alphabetical order.
```suggestion
            from sklearn.datasets import load_iris
            from sklearn.linear_model import SGDClassifier
            from sklearn.model_selection import train_test_split
```

File: optuna/pruners/median.py [hunk 2]
```diff
@@ -11,22 +11,20 @@ class MedianPruner(PercentilePruner):
...
-        .. testsetup::
+        .. testcode::
+
+            import optuna
 
             import numpy as np
+
+            from sklearn.linear_model import SGDClassifier
+            from sklearn.datasets import load_iris
             from sklearn.model_selection import train_test_split
```
Comment (toshihikoyanase) [inline optuna/pruners/median.py]: ditto.
```suggestion
            import optuna
            import numpy as np
            from sklearn.datasets import load_iris
            from sklearn.linear_model import SGDClassifier
            from sklearn.model_selection import train_test_split
```

File: optuna/pruners/nop.py [hunk 3]
```diff
@@ -11,22 +11,20 @@ class NopPruner(BasePruner):
...
-        .. testsetup::
+        .. testcode::
+
+            import optuna
 
             import numpy as np
+
+            from sklearn.linear_model import SGDClassifier
+            from sklearn.datasets import load_iris
             from sklearn.model_selection import train_test_split
```
Comment (toshihikoyanase) [inline optuna/pruners/nop.py]: ditto.
```suggestion
            import optuna
            import numpy as np
            from sklearn.datasets import load_iris
            from sklearn.linear_model import SGDClassifier
            from sklearn.model_selection import train_test_split
```

File: optuna/pruners/percentile.py [hunk 4]
```diff
@@ -74,22 +74,20 @@ class PercentilePruner(BasePruner):
...
-        .. testsetup::
+        .. testcode::
+
+            import optuna
 
             import numpy as np
+
+            from sklearn.linear_model import SGDClassifier
+            from sklearn.datasets import load_iris
             from sklearn.model_selection import train_test_split
```
Comment (toshihikoyanase) [inline optuna/pruners/percentile.py]: ditto.
```suggestion
            import optuna
            import numpy as np
            from sklearn.datasets import load_iris
            from sklearn.linear_model import SGDClassifier
            from sklearn.model_selection import train_test_split
```

File: optuna/pruners/successive_halving.py [hunk 5]
```diff
@@ -33,22 +33,20 @@ class SuccessiveHalvingPruner(BasePruner):
...
-        .. testsetup::
+        .. testcode::
+
+            import optuna
 
             import numpy as np
+
+            from sklearn.linear_model import SGDClassifier
+            from sklearn.datasets import load_iris
             from sklearn.model_selection import train_test_split
```
Comment (toshihikoyanase) [inline optuna/pruners/successive_halving.py]: ditto.
```suggestion
            import optuna
            import numpy as np
            from sklearn.datasets import load_iris
            from sklearn.linear_model import SGDClassifier
            from sklearn.model_selection import train_test_split
```

File: optuna/trial.py [hunk 6]
```diff
@@ -160,20 +160,17 @@ def suggest_uniform(self, name, low, high):
...
-                y = np.random.randint(0, 2, 200)
-                X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
-
             .. testcode::
 
                 import optuna
+                import numpy as np
+                from sklearn.datasets import load_iris
                 from sklearn.neural_network import MLPClassifier
+                from sklearn.model_selection import train_test_split
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
                from sklearn.model_selection import train_test_split
                from sklearn.neural_network import MLPClassifier
```

File: optuna/trial.py [hunk 7]
```diff
@@ -221,20 +218,17 @@ def suggest_loguniform(self, name, low, high):
...
-                y = np.random.randint(0, 2, 50)
-                X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
-
             .. testcode::
 
                 import optuna
+                import numpy as np
+                from sklearn.datasets import load_iris
                 from sklearn.svm import SVC
+                from sklearn.model_selection import train_test_split
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
                from sklearn.model_selection import train_test_split
                from sklearn.svm import SVC
```

File: optuna/trial.py [hunk 8]
```diff
@@ -286,20 +280,17 @@ def suggest_discrete_uniform(self, name, low, high, q):
...
-                X = np.random.randn(50).reshape(-1, 1)
-                y = np.random.randint(0, 2, 50)
-                X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
-
             .. testcode::
 
                 import optuna
+                import numpy as np
                 from sklearn.ensemble import GradientBoostingClassifier
+                from sklearn.datasets import load_iris
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
                from sklearn.datasets import load_iris
                from sklearn.ensemble import GradientBoostingClassifier
```

File: optuna/trial.py [hunk 9]
```diff
@@ -345,20 +336,17 @@ def suggest_int(self, name, low, high):
...
-                X = np.random.randn(50).reshape(-1, 1)
-                y = np.random.randint(0, 2, 50)
-                X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
-
             .. testcode::
 
                 import optuna
+                import numpy as np
                 from sklearn.ensemble import RandomForestClassifier
+                from sklearn.datasets import load_iris
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
                from sklearn.datasets import load_iris
                from sklearn.ensemble import RandomForestClassifier
```

File: optuna/trial.py [hunk 10]
```diff
@@ -402,20 +390,17 @@ def suggest_categorical(self, name, choices):
...
-                y = np.random.randint(0, 2, 50)
-                X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
-
             .. testcode::
 
                 import optuna
+                import numpy as np
                 from sklearn.svm import SVC
+                from sklearn.datasets import load_iris
+                from sklearn.model_selection import train_test_split
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
                from sklearn.datasets import load_iris
                from sklearn.model_selection import train_test_split
                from sklearn.svm import SVC
```

File: optuna/trial.py [hunk 11]
```diff
@@ -467,20 +452,17 @@ def report(self, value, step):
...
-                X = np.random.randn(50).reshape(-1, 1)
-                y = np.random.randint(0, 2, 50)
-                X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
-
             .. testcode::
 
                 import optuna
+                import numpy as np
                 from sklearn.linear_model import SGDClassifier
+                from sklearn.datasets import load_iris
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
                from sklearn.datasets import load_iris
                from sklearn.linear_model import SGDClassifier
```

File: optuna/trial.py [hunk 12]
```diff
@@ -160,20 +160,17 @@ def suggest_uniform(self, name, low, high):
...
 
                 import optuna
+                import numpy as np
+                from sklearn.datasets import load_iris
                 from sklearn.neural_network import MLPClassifier
+                from sklearn.model_selection import train_test_split
+
+                X, y = load_iris(return_X_y=True)
+                X_train, X_test, y_train, y_test = train_test_split(X, y)
+                classes = np.unique(y)
```
Comment (toshihikoyanase) [inline optuna/trial.py]: I don't think the `classes` variable is used in this example.
```suggestion
```

File: optuna/trial.py [hunk 13]
```diff
@@ -221,20 +218,17 @@ def suggest_loguniform(self, name, low, high):
...
 
                 import optuna
+                import numpy as np
+                from sklearn.datasets import load_iris
                 from sklearn.svm import SVC
+                from sklearn.model_selection import train_test_split
+
+                X, y = load_iris(return_X_y=True)
+                X_train, X_test, y_train, y_test = train_test_split(X, y)
+                classes = np.unique(y)
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
```

File: optuna/trial.py [hunk 14]
```diff
@@ -286,20 +280,17 @@ def suggest_discrete_uniform(self, name, low, high, q):
...
 
                 import optuna
+                import numpy as np
                 from sklearn.ensemble import GradientBoostingClassifier
+                from sklearn.datasets import load_iris
+                from sklearn.model_selection import train_test_split
+
+                X, y = load_iris(return_X_y=True)
+                X_train, X_test, y_train, y_test = train_test_split(X, y)
+                classes = np.unique(y)
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
```

File: optuna/trial.py [hunk 15]
```diff
@@ -345,20 +336,17 @@ def suggest_int(self, name, low, high):
...
 
                 import optuna
+                import numpy as np
                 from sklearn.ensemble import RandomForestClassifier
+                from sklearn.datasets import load_iris
+                from sklearn.model_selection import train_test_split
+
+                X, y = load_iris(return_X_y=True)
+                X_train, X_test, y_train, y_test = train_test_split(X, y)
+                classes = np.unique(y)
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
```

File: optuna/trial.py [hunk 16]
```diff
@@ -402,20 +390,17 @@ def suggest_categorical(self, name, choices):
...
 
                 import optuna
+                import numpy as np
                 from sklearn.svm import SVC
+                from sklearn.datasets import load_iris
+                from sklearn.model_selection import train_test_split
+
+                X, y = load_iris(return_X_y=True)
+                X_train, X_test, y_train, y_test = train_test_split(X, y)
+                classes = np.unique(y)
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
```

File: optuna/trial.py [hunk 17]
```diff
@@ -467,20 +452,17 @@ def report(self, value, step):
...
 
                 import optuna
+                import numpy as np
                 from sklearn.linear_model import SGDClassifier
+                from sklearn.datasets import load_iris
+                from sklearn.model_selection import train_test_split
+
+                X, y = load_iris(return_X_y=True)
+                X_train, X_test, y_train, y_test = train_test_split(X, y)
+                classes = np.unique(y)
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
```

File: optuna/trial.py [hunk 18]
```diff
@@ -564,21 +546,18 @@ def set_user_attr(self, key, value):
...
 
                 import optuna
+                import numpy as np
+                from sklearn.datasets import load_iris
+                from sklearn.model_selection import train_test_split
                 from sklearn.neural_network import MLPClassifier
 
+                X, y = load_iris(return_X_y=True)
+                X_train, X_test, y_train, y_test = train_test_split(X, y)
+                classes = np.unique(y)
```
Comment (toshihikoyanase) [inline optuna/trial.py]: ditto.
```suggestion
```

**Comment by toshihikoyanase** [comment]
@barneyhill Thank you for your update. Please merge master into your branch to resolve CI failures. New dependencies are added along with the hyperparameter importance feature recently.

Error log:
```console
    Running setup.py install for pyrfr ... - error
    ERROR: Command errored out with exit status 1:
     command: /home/circleci/project/venv/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-suwagejn/pyrfr/setup.py'"'"'; __file__='"'"'/tmp/pip-install-suwagejn/pyrfr/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-2jrxb5qz/install-record.txt --single-version-externally-managed --compile --install-headers /home/circleci/project/venv/include/site/python3.5/pyrfr
         cwd: /tmp/pip-install-suwagejn/pyrfr/
    Complete output (7 lines):
    running install
    running build_ext
    building 'pyrfr._regression' extension
    swigging pyrfr/regression.i to pyrfr/regression_wrap.cpp
    swig -python -c++ -modern -features nondynamic -I./include -o pyrfr/regression_wrap.cpp pyrfr/regression.i
    unable to execute 'swig': No such file or directory
    error: command 'swig' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/circleci/project/venv/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-suwagejn/pyrfr/setup.py'"'"'; __file__='"'"'/tmp/pip-install-suwagejn/pyrfr/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-2jrxb5qz/install-record.txt --single-version-externally-managed --compile --install-headers /home/circleci/project/venv/include/site/python3.5/pyrfr Check the logs for full command output.
```

**Comment by toshihikoyanase** [comment]
Hello @barneyhill,
I found that this PR was labeled as `stale` by bot. Do you have any problems?
I think the rest of the tasks are
- apply my suggestions (if they are ok), and
- rebase the branch (e.g., `git rebase master`).

Please let me know if you have any comments or questions.

**Comment by barneyhill** [comment]
Hello @toshihikoyanase,
Thanks for taking the time to review my PR!
Unfortunately I've been having several issues with my Chromebook so I've been unable to apply your suggestions.

**Comment by toshihikoyanase** [comment]
@barneyhill Sure! I'll create a new PR including your changes. Thank you for your time.

**Comment by toshihikoyanase** [comment]
I created a PR #1221 including your changes. Let me close this PR.

**Pull request closed without merging**